src/OpenTelemetry/Logs/LogRecord.cs

```diff
@@ -134,21 +141,28 @@ public void ForEachScope<TState>(Action<LogRecordScope, TState> callback, TState
         }
 
         /// <summary>
-        /// Buffers the scopes attached to the log into a list so that they can
+        /// Buffers the states &amp; scopes attached to the log so that they can
         /// be safely processed after the log message lifecycle has ended.
         /// </summary>
-        internal void BufferLogScopes()
+        internal void Buffer()
         {
-            if (this.ScopeProvider == null || this.bufferedScopes != null)
+            if (this.stateValues != null && this.bufferedStateValues == null)
```

Comment:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Logs/OpenTelemetryLogger.cs#L67 should we set to null similar to being done for scopes?

Reply:
+1